pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/actions/RestartDeclarativePipelineAction.java

```diff
@@ -107,6 +107,26 @@ public boolean isRestartEnabled() {
                 getExecution() != null;
     }
 
+    public Api getApi() {
+        return new Api(this);
+    }
+
+    @Restricted(DoNotUse.class)
+    @RequirePOST
+    public void doRestartPipeline(@QueryParameter String stageName) {
+        StageRestartResult restartResult = new StageRestartResult("success", true);
+
+        if (!isRestartEnabled()) {
```

Comment:
@oleg-nenashev Already checked the permission, but I don't know how to deal with it.